### PR TITLE
Multi directories support

### PR DESCRIPTION
**Pull request by EddyTrotin**
Added multiple directories support option. If enabled, modules should call configure method again to register new directories. Translating function (__, __mf ...) should pass directory name as argument or default directory define at configuration will be used.
Also added a clean function in the makefile and corrected one mistake.

### CODE REVIEW

**Comment by sngo25** [comment]
No documentation update?

File: README.md [hunk 1]
```diff
@@ -107,6 +107,47 @@ app.use(express.static(__dirname + '/www'));
 app.use('/en', express.static(__dirname + '/www'));
 app.use('/de', express.static(__dirname + '/www'));
 ```
+##Multiple Directories support
```
Comment (kclang) [inline README.md:110]: miss space character here and others line

File: README.md [hunk 2]
```diff
@@ -107,6 +107,47 @@ app.use(express.static(__dirname + '/www'));
...
+
+i18n-node support a multiple directory usage. This option let you register different directories to provide translations. Just add "multiDirectories : true" in your configuration then use the configure method to provide new directory and a name for it. Finally call translations methods with directory name as last argument.
+Please note that in case you do not use the updateFiles you may ommit directory name and juste call translations method as usual.
+
+###Exemple
+####In core/index.js
```
Comment (kclang) [inline README.md:116]: should provide common example, not specific in Linagora's projects

File: i18n.js [hunk 3]
```diff
@@ -69,109 +72,172 @@ module.exports = (function() {
...
-          if (typeof api[method] !== 'undefined') {
-            api[method] = alias;
+    if (!multiDirectories || opt.multiDirectories === false) {
+      // reset locales
+      locales = {};
+
+      // Provide custom API method aliases if desired
+      // This needs to be processed before the first call to applyAPItoObject()
+      if (opt.api && typeof opt.api === 'object') {
+        for (var method in opt.api) {
```
Comment (kclang) [inline i18n.js:82]: use Object.keys() instead of for..in here and in other places

File: i18n.js [hunk 4]
```diff
@@ -69,109 +72,172 @@ module.exports = (function() {
...
+        //set first directory as 'default' if name is not provided
+        if (!opt.dirName) {
+          opt.dirName = 'default';
+        }
+        directories[opt.dirName] = (typeof opt.directory === 'string') ?
+            opt.directory : path.join(__dirname, 'locales');
+        //When using multiple directories, directory is used to keep current working dir
+        //Directory will be redefined when a directory manipulation is needed
+        directory = directories[opt.dirName];
+      }else{
```
Comment (kclang) [inline i18n.js:125]: miss space character here and in others if..else

File: i18n.js [hunk 5]
```diff
@@ -22,6 +22,7 @@ var vsprintf = require('sprintf-js').vsprintf,
     require('make-plural/data/plurals.json')
   ),
   parseInterval = require('math-interval-parser').default;
+var extend = require('extend');
```
Comment (kclang) [inline i18n.js:25]: should keep the original format of author: using single 'var' declaration statement, one per line

File: i18n.js [hunk 6]
```diff
@@ -69,109 +72,172 @@ module.exports = (function() {
...
+        logError('Can not register non string path');
+        return false;
+      }
+      directories[opt.dirName] = opt.directory;
+
+      directory = directories[opt.dirName];
+
+      // when missing locales we try to guess that from directory
+      opt.locales = opt.locales || guessLocales(directory);
+      opt.locales.forEach(function(locale){
```
Comment (kclang) [inline i18n.js:212]: why this loop? it's unnecessary

File: i18n.js [hunk 7]
```diff
@@ -69,109 +72,172 @@ module.exports = (function() {
...
+        extend(locales[locale], locales[locale] ? locales[locale] : {});
+      });
+
+      // read language fallback map
+      fallbacks = (typeof opt.fallbacks === 'object') ?
+          extend(fallbacks, opt.fallbacks) : fallbacks;
 
+      // implicitly read all locales
+      if (Array.isArray(opt.locales)) {
```
Comment (kclang) [inline i18n.js:221]: this "if" block duplicates the one at line 176

**Pull request closed without merging**